File: MediCareWebApi/Models/PrescriptionMedicament.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using MediCare.Enums;

namespace MediCare.Models
{
	public class PrescriptionMedicament
	{
		[Required]
		[MaxLength(256)]
		public string Dosage { get; set; }

		[Required]
		public int Quantity { get; set; }

		[Required]
		public MedicamentUnit MedicamentUnit { get; set; }

		public string? Notes { get; set; }

		[Key]
		[Column(Order = 1)]
		public int PrescriptionAppointmentId { get; set; }

		[ForeignKey("PrescriptionAppointmentId")]
		public Prescription Prescription { get; set; }

		[Key]
		[Column(Order = 2)]
		public int MedicamentId { get; set; }

		[ForeignKey("MedicamentId")]
		public Medicament Medicament { get; set; }
	}
}

File: MediCareWebSpa/src/app/DTOs/models/prescription-medicament.ts
import { MedicamentUnit } from "../../enums/medicament-unit";
import { Medicament } from "./medicament";

export class PrescriptionMedicament {
  dosage: string;
  quantity: number;
  medicamentUnit: MedicamentUnit;
  prescriptionId: number;
  medicamentId: number;
  medicament!: Medicament;
  notes: string | null;

  constructor(
    dosage: string,
    quantity: number,
    prescriptionId: number,
    medicamentUnit: MedicamentUnit,
    medicamentId: number,
    notes: string | null
  ) {
    this.dosage = dosage
    this.quantity = quantity;
    this.prescriptionId = prescriptionId;
    this.medicamentUnit = medicamentUnit;
    this.medicamentId = medicamentId;
    this.notes = notes;
  }
}
